harness/src/network.rs

```diff
@@ -57,6 +57,8 @@ pub struct Network {
     pub storage: HashMap<u64, MemStorage>,
     /// Drop messages from `from` to `to` at a rate of `f64`.
     dropm: HashMap<Connection, f64>,
+    /// Delay sending messages from `from` to `to` at a rate of `f64` by blocking `Duration` time.
+    delaym: HashMap<Connection, (f64, Duration)>,
```

Comment:
This is a little confusing. So it does not just delay the messages in the connection, it also delays all the pending messages that may dispatch in other connection?

Reply:
That's the same concern I described above that connections shouldn't affect each other. But our `Network` seems to be single-thread based now but works fine in most functional test cases.

If I want to add some benches such as benchmarking proposing, this implementation is unsuitable. The design might be reconsidered and could be a lot of work.